src/java.base/share/classes/java/lang/foreign/SymbolLookup.java

```diff
@@ -292,6 +294,9 @@ static SymbolLookup libraryLookup(String name, Arena arena) {
     static SymbolLookup libraryLookup(Path path, Arena arena) {
         Reflection.ensureNativeAccess(Reflection.getCallerClass(),
                 SymbolLookup.class, "libraryLookup");
+        if (!path.getFileSystem().equals(FileSystems.getDefault())) {
```

Comment:
This trusts the custom provider's equals implementation, which might be okay but you can use == here instead.

I think I would probably expand the text of the existing IAE description to say that IAE is thrown if the path is not associated with the default file system or not a path to a valid library.

It's a new testable assertion so I assume we should add a new test and track the spec change with a CSR.